### ISSUE TITLE
experiments/ablations.py writes only final_loss per config; no seed/variance

### ISSUE THREAD

**Issue by sachncs** [sachncs/gradient-regularized-newton-boosting-trees#34]
## Problem

`experiments/ablations.py` writes a CSV with one row per hyperparameter configuration. Each row records `final_loss` only — no seed column, no run index, no variance estimate.

This means:

1. The CSV cannot tell whether a single `final_loss` value is stable across seeds.
2. Reproducing the published numbers requires guessing the seed (the synthetic dataset uses `seed=42` in `make_synthetic`, but this is undocumented).
3. The README claims to "systematically vary five axes" but provides no error bars.

## Evidence

- `experiments/ablations.py:38-54` — `make_synthetic(seed=42)`.
- `experiments/ablations.py:84-99` — CSV writer writes only `final_loss` per row.
- The CSV header is generated from the first row's keys; no `seed`, `run`, or `final_loss_std` columns exist.

## Root Cause

The ablation script was written for a single run per configuration. The seed is implicit in `make_synthetic`.

## Impact

- **Reproducibility**: the published ablation numbers are single-run; no confidence interval can be reported.
- **DX**: a user comparing their own results to the published ones cannot tell if a difference is within noise.

## Recommendation

Modify the ablation script to:

1. Run each configuration with multiple seeds (e.g., seeds `[42, 43, 44]`).
2. Record `seed`, `final_loss` per run, and (optionally) compute `final_loss_mean` and `final_loss_std` per configuration.
3. Either output long-form (one row per (config, seed)) or wide-form (one row per config with mean/std columns).

```python
SEEDS = [42, 43, 44]
for seed in SEEDS:
    for ...:
        x, y = make_synthetic(seed=seed)
        model = ...
        model.fit(x, y)
        rows.append({"seed": seed, "final_loss": ..., ...})
```

Then post-processing can compute mean/std.

## Acceptance Criteria

- [ ] The CSV includes a `seed` column.
- [ ] Each configuration appears at least 3 times (3 seeds).
- [ ] The README's `Reproducible Experiments` section documents the seed list.

## Verification

```bash
python experiments/ablations.py
# Inspect experiments/ablations.csv — should have ~324 rows (108 configs × 3 seeds).
```

## Dependencies

None.

## Risk

None — additive change to a script.

**Comment by sachncs** [comment]
Resolved in PR #40.
